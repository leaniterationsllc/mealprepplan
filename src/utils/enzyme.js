/**
 * Copyright (c) 2017-present, Lean Iterations LLC.
 * All rights reserved.
 *
 * This source code is licensed under the MIT-style license found in the
 * LICENSE file in the root directory of this source tree.
 */
// configuring enzyme to use the adapter for React 16
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

export default Enzyme.configure({ adapter: new Adapter() });
