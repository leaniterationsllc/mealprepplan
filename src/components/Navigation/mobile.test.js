import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Mobile from './mobile';
import enzymeconf from '../../utils/enzyme';

describe('Navigation - mobile', () => {
  it('renders', () => {
    const onToggleNav = jest.fn();
    const wrapper = shallow(
      <Mobile mobileStyle="" toggleNav={onToggleNav}>
        <div>Test</div>
      </Mobile>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('toggles the nav correctly', () => {
    const onToggleNav = jest.fn();
    const wrapper = shallow(
      <Mobile mobileStyle="" toggleNav={onToggleNav}>
        <div>Test</div>
      </Mobile>
    );

    wrapper.find('div[role="button"]').simulate('click');
    expect(onToggleNav).toHaveBeenCalled();
  });
});
