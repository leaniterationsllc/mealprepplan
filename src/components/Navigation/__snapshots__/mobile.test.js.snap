// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation - mobile renders 1`] = `
<div>
  <div
    id="content"
  >
    <nav
      className=""
    >
      <div
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      >
        <i
          className="fas fa-times-circle fa-lg"
        />
      </div>
      <div>
        Test
      </div>
    </nav>
  </div>
  <div
    data-target-tag-name="DIV"
    id="target"
  />
</div>
`;
